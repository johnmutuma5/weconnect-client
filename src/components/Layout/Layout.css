.content {
    margin-top: 70px;
    min-height: calc(100vh - 170px);
    position: relative;
    display: flex;
    justify-content: space-around;
}

.main-content {
    flex-basis: 100%;
    justify-content: space-between;
}
