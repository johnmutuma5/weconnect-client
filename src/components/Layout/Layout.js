import React from 'react';
import Aux from '../../hoc/Aux';
import Toolbar from '../Navigation/Toolbar/Toolbar';

const layout = (props) => (
    <Aux>
        <header>
            <Toolbar />
        </header>
        <div>
            SideDrawer,
            Backdrop
        </div>
        <div className='wrapper'>
            <main>
                { props.children }
            </main>
        </div>
        <br></br>
        <br></br>
        <br></br>
        <hr></hr>
        <footer>
            <div className="Copyright">
              <p className="zeroBottom"> &copy; 2018 WeConnect, All Rights Reserved.</p>
            </div>
        </footer>
    </Aux>
);

export default layout;
