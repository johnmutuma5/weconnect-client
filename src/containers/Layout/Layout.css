.content {
    margin-top: 100px;
    min-height: calc(100vh - 170px);
    position: relative;
    display: flex;
    justify-content: space-around;
}

.content .main-content {
    flex-basis: 100%;
    justify-content: space-between;
}
