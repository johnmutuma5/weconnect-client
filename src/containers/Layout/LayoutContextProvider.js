import React from 'react';
import { PropTypes } from 'prop-types';

class LayoutContextProvider extends React.Component {
    getChildContext() {
        return {
            state: this.props.state,
            handleToggleSideDrawer: this.props.handleToggleSideDrawer,
            toggleUserGettingStarted: this.props.toggleUserGettingStarted
        }
    }

    render() {
        return this.props.children;
    }
}

LayoutContextProvider.childContextTypes = {
    state: PropTypes.object.isRequired,
    handleToggleSideDrawer: PropTypes.func.isRequired,
    toggleUserGettingStarted: PropTypes.func.isRequired

}

export default LayoutContextProvider;
