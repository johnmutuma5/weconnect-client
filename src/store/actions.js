const getBusinesses = (businesses) => (
    {
        type: 'GET_BUSINESSES',
        payload: businesses
    }
);


export default {
    getBusinesses: getBusinesses
}
