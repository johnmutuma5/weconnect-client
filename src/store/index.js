import { events } from './events';
import ResourcesProvider from './ResourcesProvider';


class Store {
    constructor(reducer) {
        this.reducer = reducer;
        this.events = events
    }

    dispatch(action) {
        this.state = this.reducer(this.state, action);
        // notify listeners that this event has occured and updated state
        // eventTypes are identified by action types
        console.log(this.state);
        this.notify(action.type);
    }

    after(eventType, listener) {
        // this will be used to subscribe listeners to various actions
        // eventTypes are identified by action types
        try {
            this.events[eventType].push(listener);
        } catch (e) {
            console.log(`Ensure you have registered ${ eventType } in the events file`);
        }
        // return a function to unsubscribe the listener
        const unsubscribe = () => {
            this.events[eventType] = this.events[eventType]
                    .filter((_listener) => _listener !== listener);
        };
        return unsubscribe;
    }

    subscribe(subscriptions, listener) {
        // subscription to store events return an array of each subscription's
        // revoker
        let subscriptionsRevokers = []
        for(let subscription of subscriptions){
            const unsubscribe = this.after(subscription, listener);
            subscriptionsRevokers.push(unsubscribe);
        }
        return subscriptionsRevokers;
    }

    notify(eventType) {
        // this will be used to notify listners when events of actionType occur
        for (let listner of this.events[eventType])
            listner(this.state);
    }
}


export const createStore = (reducer) => new Store(reducer);
export default ResourcesProvider;
