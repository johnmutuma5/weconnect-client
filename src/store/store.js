import { events } from './events';


class Store {
    constructor(reducer) {
        this.reducer = reducer;
        this.events = events
    }

    dispatch(action) {
        this.state = this.reducer(undefined, action);
        this.notify(action.type);
    }

    after(actionType, listener) {
        // this will be used to subscribe listeners to various actions
        this.events[actionType].push(listener);
        // return a function to unsubscribe the listener
        const unsubscribe = () => {
            this.events[actionType] = this.events[actionType]
                    .filter((_listener) => _listener !== listener);
        };
        return unsubscribe;
    }

    notify(actionType) {
        // this will be used to notify listners when events of actionType occur
        for (let listner of this.events[actionType])
            listner(this.state);
    }
}


export const createStore = (reducer) => new Store(reducer);
