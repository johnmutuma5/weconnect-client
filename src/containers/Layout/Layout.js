import React from 'react';
import PropTypes from 'prop-types';
import LayoutContextProvider from './LayoutContextProvider';
import Navigation from '../../components/Navigation/Navigation';
import GettingStarted from '../Auth/GettingStarted';
import Backdrop from '../../components/UI/Backdrop/Backdrop';
import Footer from './Footer/Footer';
import { initLayoutState, toggleGettingStarted } from '../../store/actions/actions';

import './Layout.css';


class Layout extends React.Component {
    constructor(props, context){
        super();
        this.store = context.store;
        this.state = this.store.layoutState;
        this.subscriptions = [
            'INIT_LAYOUT_STATE',
            'LOGIN_USER',
            'LOGOUT_USER',
            'TOGGLE_SIDE_DRAWER',
            'TOGGLE_GETTING_STARTED',
            'TOGGLE_REGISTERING_BUSINESS'
        ];
    }

    componentWillMount() {
        this.subscriptionsRevokers =
            this.store.subscribe(this.subscriptions, this.handleStateDidUpdate.bind(this));
        this.store.dispatch(initLayoutState());
    }

    componentWillUnmount() {
        for(let subscriptionRevoker of this.subscriptionsRevokers)
            subscriptionRevoker();
    }

    handleStateDidUpdate(state) {
        this.setState(state.layoutState);
    }

    render() {
        const store = this.context.store;

        // prepare to render getting started elements
        let gettingStarted = null;
        if(this.state.userGettingStarted)
            gettingStarted =
                <GettingStarted visible={this.state.userGettingStarted} />

        return (
            <LayoutContextProvider layoutState={ this.state } >
                <Navigation />
                <div className="content">
                    <div className='container main-content'>
                        <main>
                            { this.props.children }
                        </main>
                    </div>
                    { gettingStarted }
                    <Backdrop
                        active={this.state.userGettingStarted}
                        id={ 'GettingStartedBackdrop' }
                        click={ () => store.dispatch(toggleGettingStarted()) }/>
                    <Footer />
                </div>
            </LayoutContextProvider>
        );
    }
}

Layout.contextTypes = {
    store: PropTypes.object.isRequired
}


export default Layout;
