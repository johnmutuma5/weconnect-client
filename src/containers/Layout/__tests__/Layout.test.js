import Layout from '../Layout';
import { createStore } from '../../../store';
import weConnectReducer from '../../../store/baseReducer';

describe('<Layout /> Component', () => {
    it('renders correctly', () => {
        const store = createStore(weConnectReducer);
        const context = { store: store };
        const wrapper = shallow(<Layout />, { context });
        expect(wrapper).toMatchSnapshot();
    });
})
