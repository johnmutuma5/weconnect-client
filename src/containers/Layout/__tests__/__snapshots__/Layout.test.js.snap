// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Component renders correctly 1`] = `
<aux>
  <mainNavigation />
  <div
    className="content"
  >
    <div
      className="container main-content"
    >
      <main />
    </div>
    <backDrop
      active={false}
      click={[Function]}
      id="GettingStartedBackdrop"
    />
  </div>
</aux>
`;
